### PR TITLE
feat(angular): add support for standalone applications

### PR DESCRIPTION
**Pull request by skoropadas**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

At the current moment it's not possible to add providers to standalone bootstrap function.

## What is the new behavior?

I have added the following features:
- Added the `getBootstrapApplicationFn` function to allow obtaining the bootstrap function.
- Added the `addProviderToBootstrapApplicationFn` function for adding a new provider to the bootstrap function.
- Refactored the `getBootstrapFn` function, it can now return `undefined` if the function was not found.
- Refactored a couple of internal helpers to avoid code duplication for the new functions.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ng-morph/ng/bootstrap/add-provider-to-bootstrap-application-fn.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { CallExpression } from "ts-morph";
+import { pushToObjectArgument } from "../helpers/push-to-object-argument";
+
+export function addProviderToBootstrapApplicationFn(
+  callExpression: CallExpression,
+  provider: string,
+  { unique = false }: { unique?: boolean } = {}
+) {
+  const [_, options = callExpression.addArgument(`{providers: []}`)] = callExpression.getArguments();
```
Comment (vladimirpotekhin) [inline libs/ng-morph/ng/bootstrap/add-provider-to-bootstrap-application-fn.ts:9]: btw in Taiga, we found that in some projects they write like this:

```
import {OPTIONS} from "./options"

bootstrapApplication(AppComponent, OPTIONS)
```

(second argument is not object, but constant)

we handled it like this:

```
const optionObject = getOptionObject(options)

function getOptionsObject(
    options: Identifier | ObjectLiteralExpression,
): ObjectLiteralExpression {
    if (Node.isObjectLiteralExpression(options)) {
        return options;
    }

    const definition = options.getDefinitionNodes()[0];

    return definition.getChildrenOfKind(SyntaxKind.ObjectLiteralExpression)[0];
}

...
```

Comment (skoropadas) [reply libs/ng-morph/ng/bootstrap/add-provider-to-bootstrap-application-fn.ts:9]: I think it's a good think for support, I'll add it, thanks!

File: libs/ng-morph/ng/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-export * from './helpers/get-bootstrap-fn';
 export * from './helpers/get-main-module';
```
Comment (MarsiBarsi) [inline libs/ng-morph/ng/index.ts:1]: let's export helpers as well, they can be useful for someone

File: libs/ng-morph/ng/bootstrap/get-bootstrap-fn.ts [hunk 3]
```diff
@@ -1,23 +1,20 @@
-import { CallExpression, Node } from 'ts-morph';
+import { CallExpression, Node, SyntaxKind } from 'ts-morph';
 import { getImports } from 'ng-morph/imports';
-import { getActiveProject } from 'ng-morph/project';
 
-export function getBootstrapFn(mainFilePath: string): CallExpression {
+export function getBootstrapFn(mainFilePath: string): CallExpression | undefined {
```
Comment (MarsiBarsi) [inline libs/ng-morph/ng/bootstrap/get-bootstrap-fn.ts:4]: I see it as a breking change because it can cause TS errors in old scripts after update

**Review by MarsiBarsi — COMMENTED** [review]
Great, just added a couple comments

File: libs/ng-morph/ng/bootstrap/get-bootstrap-fn.ts (hunk 3 above)
Comment (skoropadas) [reply libs/ng-morph/ng/bootstrap/get-bootstrap-fn.ts:4]: If strict mode is enabled for schematics then probably yes it's a breaking change, but I believe it's how it should work after standalone apps release

File: libs/ng-morph/ng/index.ts (hunk 2 above)
Comment (sevaru) [reply libs/ng-morph/ng/index.ts:1]: My comment is slightly out of scope of this PR,
But if we plan to make them more "public" (by providing them in index) 
I'd like to suggest optional naming adjustments for better clarity:

`push-to-array-property` -> `push-to-decorator-array-property`.
`push-to-object-argument` -> `push-to-object-literal-property`

These ideas are optional :)

Comment (skoropadas) [reply libs/ng-morph/ng/index.ts]: Exported and renamed to make it clear what they do

**Pull request merged**